### PR TITLE
Fdtgrep and libfdt_extra

### PR DESCRIPTION
**Pull request by sjg20**
This is the long-awaited fdtgrep tool with the libfdt additions to make it work.

These changes were rejected from the main libfdt tree, so are housed here.

### CODE REVIEW

File: License/README [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+License headers in the source files use SPDX. See Linux Foundation's
+SPDX project [1].
+
+If a "SPDX-License-Identifier:" line references more than one Unique
+License Identifier, then this means that the respective file can be
+used under the terms of either of these licenses, i. e. with
+
+	SPDX-License-Identifier: GPL-2.0+ OR BSD-3-Clause
+
+you can choose between GPL-2.0+ and BSD-3-Clause licensing.
```
Comment (robherring) [inline License/README]: Here you say 3 clause, but the license file is 2 clause.

File: Makefile
Comment (robherring) [inline Makefile:1]: dtc is trying to move to meson. Can't we do that from the start here? Then we're not trying to maintain compatibility when meson support is added.

Comment (sjg20) [reply Makefile:1]: Hmm I was wondering about that...I will take a look

File: License/README (hunk 1 above)
Comment (sjg20) [reply License/README]: fixed

File: Makefile
Comment (sjg20) [reply Makefile:1]: done

**Comment by sjg20** [comment]
I notice that meson tests don't work in the dtc repo

**Comment by sjg20** [comment]
Just checking if there is anything else to do here?

**Comment by robherring** [comment]
My suggestion was not to have both meson and Makefiles, but just meson. Maintaining both is a pain depending to what extent you care about both supporting all the same features/setups. But it's your repo. If you are good with it, merge it.

**Comment by sjg20** [comment]
That makes more sense...but did you see my question? How do I run the tests using meson? i.e. the equivalent of 'make check'

**Comment by robherring** [comment]
This works for me:
```
meson setup build
cd build
meson compile
meson test -v

```
It doesn't look like the result counting works right though. It seems to be a single pass or fail.

**Comment by sjg20** [comment]
OK, updated, thanks for helping with the tests

**Comment by robherring** [comment]
LGTM. Maybe merge it into 'main' instead of master?

**Comment by sjg20** [comment]
Hmm not sure how to change that, but I should be able to do both

**Comment by robherring** [comment]
There's a GH UI thing to rename master. It will redirect master to main on the web, but can't migrate users that have cloned it. So best to do that before anyone clones it (besides you).

**Comment by sjg20** [comment]
OK thanks, I think I figured that out

**Pull request merged**